R/carehomes.R

```diff
@@ -254,26 +259,44 @@ carehomes_compare <- function(state, prev_state, observed, pars) {
   model_admitted <- state["admitted", ] - prev_state["admitted", ]
   model_new <- state["new", ] - prev_state["new", ]
   model_new_admitted <- model_admitted + model_new
-  model_sero_prob_pos <- state["sero_prob_pos", ]
+  model_sero_pos <- state["sero_pos", ]
   model_sympt_cases <- state["sympt_cases", ] - prev_state["sympt_cases", ]
+  model_sympt_cases_over25 <- state["sympt_cases_over25", ] -
+    prev_state["sympt_cases_over25", ]
   model_react_pos <- state["react_pos", ]
 
-  ## Add some small exponential noise in case the number of model cases is 0
-  pillar2_pos <- model_sympt_cases +
-    rexp(length(model_sympt_cases), pars$observation$exp_noise)
-  pillar2_neg <- pars$prop_noncovid_sympt *
-    (sum(pars$N_tot) - model_sympt_cases)
-  model_pillar2_prob_pos <- (pars$pillar2_sensitivity * pillar2_pos +
-                               (1 - pars$pillar2_specificity) * pillar2_neg) /
-    (pillar2_pos + pillar2_neg)
-
-  ## Note that for REACT we exclude group 1 (0-4) and 19 (CHR)
-  react_pos <- model_react_pos +
-    rexp(length(model_react_pos), pars$observation$exp_noise)
-  react_neg <- sum(pars$N_tot[2:18]) - model_react_pos
-  model_react_prob_pos <- (pars$react_sensitivity * react_pos +
-                             (1 - pars$react_specificity) * react_neg) /
-    (react_pos + react_neg)
+  ## calculate test positive probabilities for the various test data streams
+  ## Pillar 2
+  pillar2_negs <- pars$prop_noncovid_sympt * (sum(pars$N_tot)
+                                              - model_sympt_cases)
+  model_pillar2_prob_pos <- test_prob_pos(model_sympt_cases,
+                                          pillar2_negs,
+                                          pars$pillar2_sensitivity,
+                                          pars$pillar2_specificity,
+                                          pars$observation$exp_noise)
+
+  ## Pillar 2 over 25s
```

Comment:
Should we be including CHR here in general - would come under pillar 1 testing? We exclude them from REACT so we should check if pillar 2 should take the same approach